Login_Form/views.py

```diff
@@ -1,29 +1,45 @@
 from rest_framework.views import APIView
 from rest_framework.response import Response
 from rest_framework import status
-from django.contrib.auth import authenticate
+from werkzeug.security import generate_password_hash, check_password_hash
 from .models import User
+from Login.settings import SECRET_KEY
+from jwt import encode
+import re
 from .serializers import userProfileSerializer
 
+
 class RegisterView(APIView):
     def post(self, request, *args, **kwargs):
         serializer = userProfileSerializer(data=request.data)
-        username=request.data.get('username')
-        password=request.data.get('password')
-        user = User.objects.filter(username=username,password=password).exists()
-        if user is not None:
-            return Response({'message':'user already exists'})
+        
+        if serializer.is_valid():
+            password=serializer.data['password']
+
+            if len(password) < 8:
+                return Response({'message':'Make sure your password is at lest 8 letters'}) 
+            elif re.search('[0-9]',password) is None:
+                return Response({'message':'Make sure your password has a number in it'})
+            elif re.search('[A-Z]',password) is None: 
+                return Response({'message':'Make sure your password has a capital letter in it'})
+            elif re.search('[^a-zA-Z0-9]',password) is None:
+                return Response({'message':'Make sure your password has a special character in it'}) 
+            elif password!=serializer.data['confirm_password']:
+                return Response({'message':'password not match'})
+            user=User(username=serializer.data['username'],password=generate_password_hash(password),email=serializer.data['email'])
+            user.save()
+            return Response({'message':'Register successful'}, status=status.HTTP_201_CREATED)
         else:
-            if serializer.is_valid():
-                serializer.save()
-                return Response(serializer.data, status=status.HTTP_201_CREATED)
+            return Response({'message':'already exists'},status=status.HTTP_400_BAD_REQUEST)
         
 class LoginView(APIView):
     def post(self, request, *args, **kwargs):
+        serializer = userProfileSerializer(data=request.data)
         username=request.data.get('username')
         password=request.data.get('password')
-        user = User.objects.filter(username=username,password=password).exists()
-        if user is not None:
-            return Response({'message':'Login successful'},status=status.HTTP_202_ACCEPTED)
+        user = User.objects.filter(username=username).first()
```

Comment:
why are you using filter in this?

Reply:
To get that particular user data from database if it exists so that i can compare the user entered password with the hash password